### PR TITLE
repo: VS Code Devcontainer Configuration

### PR DESCRIPTION
**Pull request by atoff**

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM webdevops/php-apache-dev:7.4
+
+ENV WEB_DOCUMENT_ROOT=/app/nexus/public
+
+WORKDIR /app/nexus/
+
+RUN curl -sL https://deb.nodesource.com/setup_16.x | bash -
+RUN apt-get install -y nodejs
+
+RUN npm install -g yarn
```
Comment (AxonC) [inline .devcontainer/Dockerfile]: line end

**Comment by AxonC** [comment]
Does this mount the live 'code' when you make a change locally in the container?

**Comment by atoff** [comment]
> Does this mount the live 'code' when you make a change locally in the container?

Yep, it mounts a volume in the docker compose file (it mounts the root of the project to `/app/nexus`)

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "mrmlnc.vscode-apache",
+    "bradlc.vscode-tailwindcss",
+    "bmewburn.vscode-intelephense-client",
+    "esbenp.prettier-vscode"
+  ],
+
+  // For use with PHP or Apache (e.g.php -S localhost:8080 or apache2ctl start)
+  "forwardPorts": [8080, 3306],
+
+  "postCreateCommand": "composer install --no-interaction && yarn"
```
Comment (AxonC) [inline .devcontainer/devcontainer.json:24]: Suppose you could run a migrate or key generate for this? maybe just leave that up to the user

**Comment by AxonC** [comment]
The only other comment I would have maybe has a README entry for how to get this all set up and running, as well as potential other options to run this locally

**Review by CalumTowers — COMMENTED** [review]
Generally indifferent to this - given it is a dev dependency. We can modify as we go if this setup isn’t exactly right.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (atoff) [reply .devcontainer/devcontainer.json:24]: I think the one-time stuff is best left up to user for now, but there is lots of scope for automating the setup here

**Review by AxonC — APPROVED** [review]

**Pull request merged**